client/lib/local-storage-helper/index.ts

```diff
@@ -0,0 +1,46 @@
+class LocalStorageHelper {
+	private static instance: LocalStorageHelper;
+	private cache: Map< string, any >;
+
+	private constructor() {
+		this.cache = new Map();
+	}
+
+	public static getInstance(): LocalStorageHelper {
+		if ( ! LocalStorageHelper.instance ) {
+			LocalStorageHelper.instance = new LocalStorageHelper();
+		}
+		return LocalStorageHelper.instance;
+	}
+
+	public getItem( key: string ): string | null {
+		if ( this.cache.has( key ) ) {
+			return this.cache.get( key );
+		}
+
+		const value = localStorage.getItem( key );
+		if ( value ) {
+			this.cache.set( key, value );
+			return value;
+		}
+
+		return null;
+	}
+
+	public setItem( key: string, value: string ): void {
```

Comment:
we might want to serialize the value and provide a more general interface with for example, `JSON.stringify`.